### PR TITLE
[zk-token-sdk] Refactor `zk-token-elgamal` pod types

### PR DESCRIPTION
**Pull request by samkim-crypto**
#### Problem
The `zk-token-elgamal` submodule contains pod types for various types in `zk-token-sdk`. Currently, all pod types live in the single `pod.rs`, which could be broken into separate submodules.

#### Summary of Changes
Break up the pod types into separate submodules within `zk-token-elgamal`.

This addresses step 1 of https://github.com/solana-labs/solana/issues/31813,

Fixes #

### CODE REVIEW

File: zk-token-sdk/src/zk_token_elgamal/pod/sigma_proofs.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+// Add the marker traits manually because `bytemuck` only adds them for some `u8` arrays
+unsafe impl Zeroable for CiphertextCommitmentEqualityProof {}
+unsafe impl Pod for CiphertextCommitmentEqualityProof {}
+
+/// Serialization of `CtxtCtxtEqualityProof`
+#[derive(Clone, Copy)]
+#[repr(transparent)]
+pub struct CiphertextCiphertextEqualityProof(pub [u8; 224]);
+
+// `CtxtCtxtEqualityProof` is a Pod and Zeroable.
```
Comment (CriesofCarrots) [inline zk-token-sdk/src/zk_token_elgamal/pod/sigma_proofs.rs:18]:
```suggestion
// `CiphertextCiphertextEqualityProof` is a Pod and Zeroable.
```

File: zk-token-sdk/src/zk_token_elgamal/pod/sigma_proofs.rs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+#[derive(Clone, Copy)]
+#[repr(transparent)]
+pub struct CiphertextCommitmentEqualityProof(pub [u8; 192]);
+
+// `CiphertextCommitmentEqualityProof` is a Pod and Zeroable.
+// Add the marker traits manually because `bytemuck` only adds them for some `u8` arrays
+unsafe impl Zeroable for CiphertextCommitmentEqualityProof {}
+unsafe impl Pod for CiphertextCommitmentEqualityProof {}
+
+/// Serialization of `CtxtCtxtEqualityProof`
```
Comment (CriesofCarrots) [inline zk-token-sdk/src/zk_token_elgamal/pod/sigma_proofs.rs:13]:
```suggestion
/// Serialization of `CiphertextCiphertextEqualityProof`
```

**Review by CriesofCarrots — COMMENTED** [review]
All the module moves look good! Thanks for doing them in separate commits; made my job easy.
Can we sneak in these 2 comment abbreviation fixes?

File: zk-token-sdk/src/zk_token_elgamal/pod/mod.rs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+mod sigma_proofs;
+
+use {
+    crate::zk_token_proof_instruction::ProofType,
+    num_traits::{FromPrimitive, ToPrimitive},
+    solana_program::instruction::InstructionError,
+};
+pub use {
+    auth_encryption::AeCiphertext,
+    bytemuck::{Pod, Zeroable},
```
Comment (CriesofCarrots) [inline zk-token-sdk/src/zk_token_elgamal/pod/mod.rs:15]: Incidentally, why do we re-export these traits?

**Review by CriesofCarrots — APPROVED** [review]

File: zk-token-sdk/src/zk_token_elgamal/pod/mod.rs (hunk 3 above)
Comment (samkim-crypto) [reply zk-token-sdk/src/zk_token_elgamal/pod/mod.rs:15]: The bytemuck version on the monorepo and spl side do not always match, so I had some issues before without re-exporting these traits. I can't seem to reproduce it or recall what the precise issue was right now though ... :confused:

**Pull request merged**